packages/core/src/utils/assertDuplicates.ts

```diff
@@ -0,0 +1,23 @@
+import type { ProofRequest } from '../modules/proofs/models/ProofRequest'
+
+import { AriesFrameworkError } from '../error/AriesFrameworkError'
+
+import { attributesToArray, predicatesToArray } from './toArray'
+
+export function checkProofRequestForDuplicates(proofRequest: ProofRequest) {
+  const attributes = attributesToArray(proofRequest)
+  const predicates = predicatesToArray(proofRequest)
+  assertDuplicatesInArray(attributes)
+  assertDuplicatesInArray(predicates)
+  assertDuplicatesInArray(attributes.concat(predicates))
```

Comment:
I think we can get away with only the last check. We check for any duplicates across the two arrays, making the first two checks redundant

Reply:
Very good point.